packages/plugin-http-client/src/rest/mapper/EntityMapper.ts

```diff
@@ -0,0 +1,29 @@
+import { Mapper } from './Mapper';
+import { AbstractEntity } from '../AbstractEntity';
+
+export class EntityMapper extends Mapper {
+  #EntityClass: AbstractEntity;
+
+  constructor(entity: AbstractEntity) {
+    super();
+
+    this.#EntityClass = entity;
+  }
+
+  deserialize(data: any) {
+    if (!data || typeof data === 'string' || data instanceof AbstractEntity) {
+      return data;
+    }
+
+    // @ts-ignore
+    return Reflect.construct(this.#EntityClass, [data]);
```

Comment:
tady generika nefunguje?

Reply:
Zrušil jsem ten ts-ignore - přidal jsem EntityConstructor interface. Ale nevim, jak to myslíš s tou generikou tady?